### ISSUE TITLE
@JS produces sub-optimal dart2js code for external getters

### ISSUE THREAD

**Issue by yjbanov** [dart-lang/sdk#42522]
For the following JS-interop code:

```dart
@JS('window.flutter_canvas_kit')
external CanvasKit get canvasKit2;

@JS()
class CanvasKit {
  external CanvasKitBlendMode get BlendMode;
}

@JS()
class CanvasKitBlendMode {
  external CanvasKitBlendModeValue get Clear;
}

@JS()
class CanvasKitBlendModeValue {
  external int get value;
}
```

`dart2js` compiles expression `canvasKit2.BlendMode.Clear.value` to the following:

```javascript
J.get$value$x(J.get$Clear$x(J.get$BlendMode$x(self.window.flutter_canvas_kit)))
```

I expected the following output:

```
self.window.flutter_canvas_kit.BlendMode.Clear.value
```

/cc @rakudrama

**Comment by sigmundch** [comment]
Yes! We'd love to have this perf improvement.

(a) we think we can inline most of thse calls
(b) we also want to add support for declaring external static extension methods that get compiled away entirely. In that case, you'd hopefully end up with code similar to the handwritten JS.

Meanwhile, as a short-term workaround, would it work if you try to use JS for the full path to the property? That is:
```dart
@JS('window.flutter_canvas_kit.BlendMode.Clear.value')
external int get canvasKitBlendModeValue;
```

I believe we should be able to generate that as a direct reference in this case.

**Comment by yjbanov** [comment]
Right, the "flat" numbers in the benchmark results were obtained using `@JS('window.flutter_canvas_kit.BlendMode.Clear.value')`.

I'm not yet sure in how many cases this workaround will work, but I'll try it.

**Comment by sigmundch** [comment]
> the "flat" numbers in the benchmark results were obtained using

Ah, sorry I missed that from your original post.
